### ISSUE TITLE
Where to go to learn...

### ISSUE THREAD

**Issue by MakoWish** [FreeRDP/FreeRDP#3878]
I have to admit, as a Systems Admin I have been VBScript'ing for many years, but I am just starting to learn VB.net. I am in the process of creating an application for managing Microsoft's Hyper-V. If anybody has used 5Nine Manager before, it is essentially like that, but the idea is to be more "Enterprise" than their solution (picture VCenter for VMware).

I have a pretty good product going so far, but I need a way to view the virtual machine consoles. FreeRDP seems to be the way to go (5Nine uses it for their console access), but I know ZERO about programming in C. 

I do not expect anybody here to create a build for me, but I would like to know, where should I go to get started with learning it? I essentially would like to embed a borderless, no title bar, FreeRDP into a WinForm, but I do not know where to begin. Trying to launch a prebuilt FreeRDP and setting the parent to the form is not working. 

Any resources or documents you could point me to would be much appreciated.

**Comment by byteboon** [comment]
It doesn't sound like you want to start here, as it would be a rather large learning curve for you. You may do better using the Microsoft supplied RDP ActiveX viewer control, which you can use with VBScript or other COM-friendly languages.

https://msdn.microsoft.com/en-us/library/aa383541(v=vs.85).aspx

**Comment by akallabeth** [comment]
Closing as off topic (and already answered)

**Comment by MakoWish** [comment]
Unfortunately, it looks like all those samples have been removed. This sounds like a good idea, and thank you for pointing it out, but I cannot find anything with working links.

I also found these:

https://blogs.msdn.microsoft.com/taylorb/2010/06/01/writing-your-own-vmconnect-appweb-interface-for-vm-console-access-to-hyper-v-vms/
https://blogs.msdn.microsoft.com/virtual_pc_guy/2010/02/03/hyper-v-activex-rdp-control/

But again, the code.msdn.microsoft.com links are all broken. :-(

**Comment by byteboon** [comment]
The codeproject link I provided still works (I just tried it). There should be enough info in there to get you started.

There's also these (code availability verified)
https://github.com/raddeus/RDP_Connection_Manager/tree/master/RDPManager
https://code.msdn.microsoft.com/CCS-LABS-Remote-Desktop-dd1f1cdf/sourcecode?fileId=74600&pathId=513223851

**Comment by MakoWish** [comment]
I may just be stuck with a large learning curve, because every sample I can actually find will not build on .NET 4.0 or higher (my project is on .NET 4.6). I also found this:

"Unfortunately, there are no plans to release the RDP Active-X control again." - DDTK_Support_Team
https://social.technet.microsoft.com/Forums/office/en-US/41da3097-94b5-4014-8ece-8f1589bdb4e7/where-to-download-hyperv-rdp-activex-control-?forum=winserverhyperv

It looks like this may be a dead-end.

**Comment by byteboon** [comment]
Here is an active C# project that is still using that control... https://github.com/terminals-Origin/Terminals

...so it seems to be technically possible, but you're going to have to do the work to understand how to add and use the control to your own project rather than just compiling someone else's code.

**Comment by MakoWish** [comment]
Thank you for all the assistance, byteboon, but it appears the answer was built into FreeRDP this whole time. I cannot seem to get the FreeRDP window to launch in a different size (I have tried "... /w:1024 /h:768" and "... /size:1024x768"), and it launches the same size every time no matter what. This sent me to the Wiki to ensure I had the syntax correct, I still can't size the window, but low and behold, I find...

/parent-window:<windowID>

This works perfectly!

Dim proc As New Process
Dim startInfo As New StartInfo With {
	.FileName = """" & appPath & "\console\wfreerdp.exe""",
	.Arguments = "/parent-window:" & Form1.Panel1.Handle.ToString() & " /u:UserName /p:........"
}
proc = Process.Start(startInfo)

I am having an issue setting keyboard focus to the FreeRDP window as a child now, but I have successfully docked it in my WinForm. That's what I needed! 👍
